XCTestBootstrap/TestManager/FBTestManagerTestReporterTestCase.h

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "FBTestManagerTestReporter.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class FBTestManagerTestReporterTestCaseFailure;
+
+/**
+ A Test Reporter Test Case.
+ */
+@interface FBTestManagerTestReporterTestCase : NSObject
+
+/**
+ Constructs a Test Reporter Test Case.
+
+ @param testClass the test class name.
+ @param method the test method name.
+ @return a new test case instance.
+ */
++ (instancetype)withTestClass:(NSString *)testClass method:(NSString *)method;
+
+/**
+ The test report status.
+ */
+@property (nonatomic) FBTestReportStatus status;
+
+/**
+ The test case duration.
+ */
+@property (nonatomic) NSTimeInterval duration;
```

Comment:
Make this externally `readonly` if it's not for consumers to modify. You can add a method like `finishWithDuration:status:` that can instead be called from `FBTestManagerTestReporterJUnit`. This can place additional requirements that the  `finishWithDuration:status:` is called only once per-instance.